src/Octoshift/HttpDownloadService.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+[assembly: InternalsVisibleTo("OctoshiftCLI.Tests")]
+namespace OctoshiftCLI
+{
+    public class HttpDownloadService
+    {
+        internal Func<string, string, Task> WriteToFile = async (path, contents) => await File.WriteAllTextAsync(path, contents);
+
+        private readonly HttpClient _httpClient;
+
+        public HttpDownloadService(HttpClient httpClient) => _httpClient = httpClient;
+
+        public virtual async Task Download(string url, string file)
+        {
+            using var response = await _httpClient.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
```

Comment:
I don't think we need the 2nd argument here. Unless you put it here for a specific reason. It looks like GetAsync(url) would work to me.

Reply:
Good call! Removed in 88f1a5efaa525a9b118c08df30df99b4435ce70e!

Reply:
Good call!  Removed in 88f1a5efaa525a9b118c08df30df99b4435ce70e!